server/src/com/thoughtworks/go/server/service/EntityHashingService.java

```diff
@@ -117,6 +124,18 @@ private String cacheKey(Object domainObject, String name) {
         return getClass(domainObject) + "." + name;
     }
 
+    private String getFromCache(String cacheKey, Object dbObject) {
+        String cachedMD5 = getFromCache(cacheKey);
+
+        if (cachedMD5 != null) {
+            return cachedMD5;
+        }
+        String md5 = CachedDigestUtils.md5Hex(new GsonBuilder().create().toJson(dbObject));
```

Comment:
The json for a plugin settings object will be something like this -
```
{"pluginId":"foo","settingsMap":[{"configKey":{"name":"k1","errors":{}},"configValue":{"value":"v1","errors":{}},"cipher":{"cipherProvider":{"environment":{"defaultDbPort":"9153","debug":false},"key":[50,54,57,50,57,56,98,99,51,49,99,52,52,54,50,48]}},"configErrors":{}}],"hasErrors":false,"errors":{}}
```
So, the things that a user will be in control of providing will be the key value pairs and that will result in different jsons each time. I think that should be good enough.